codechain_primitives/UnsignedInteger.py

```diff
@@ -0,0 +1,140 @@
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if type(args[0]) is str:
+            result = super().__new__(cls, *args, **kwargs, base=0)
+        else:
+            result = super().__new__(cls, *args, **kwargs)
+
+        if result < 0:
+            raise ValueError("Integer underflow")
+        if hasattr(cls, 'MAX_VALUE') and result > cls.MAX_VALUE:
+            raise ValueError("Integer overflow")
+        return result
+
+    def plus(self, rhsValue):
+        return self.__class__(int(self) + int(rhsValue))
+
+    def minus(self, rhsValue):
+        return self.__class__(int(self) - int(rhsValue))
+
+    def times(self, rhsValue):
+        return self.__class__(int(self) * int(rhsValue))
+
+    def idiv(self, rhsValue):
+        if int(rhsValue) < 0:
+            raise ValueError("Negative division")
+        return self.__class__(int(self) // int(rhsValue))
+
+    def mod(self, rhsValue):
+        if int(rhsValue) < 0:
+            raise ValueError("Negative modulo")
+        return self.__class__(int(self) % int(rhsValue))
+
+    def __add__(self, rhsValue):
+        return self.plus(rhsValue)
+
+    def __sub__(self, rhsValue):
+        return self.minus(rhsValue)
+
+    def __mul__(self, rhsValue):
+        return self.times(rhsValue)
+
+    def __floordiv__(self, rhsValue):
+        return self.idiv(rhsValue)
+
+    def __mod__(self, rhsValue):
+        return self.mod(rhsValue)
+
+    @classmethod
+    def from_rlp(cls, buffer):
+        if type(buffer) is not bytes and type(buffer) is not bytearray:
+            raise ValueError("Argument should be bytearray")
+        data = buffer[1:]
+        first = buffer[0]
+        if first < 0x80:
+            return cls(hex(first))
+
+        length = first - 0x80
+        if len(data) != length:
+            raise ValueError("Invalid data for Unsigned integer")
+        elif length > len(bytes.fromhex(hex(cls.MAX_VALUE)[2:])):
+            raise ValueError(
+                f"Data for Unsigned integer must be less than or equal to {len(bytes.fromhex(hex(cls.MAX_VALUE)[2:]))}")
+        elif length == 0:
+            return cls('0')
+        return cls(int.from_bytes(data, byteorder='big'))
+
+    @classmethod
+    def check_string(cls, param):
+        if type(param) is not str:
+            return False
+        try:
+            value = int(param, 0)
+            return value >= 0 and value <= cls.MAX_VALUE
+        except:
+            return False
+
+    @classmethod
+    def check(cls, param):
+        if type(param) is str:
+            return cls.check_string(param)
+        else:
+            if not isinstance(param, int):
+                return False
+        if param >= 0 and param <= cls.MAX_VALUE:
+            return True
+        return False
+
+    def to_encode_object(self):
+        result = hex(int(self))[2:]
+        result = "0" + result if len(result) % 2 == 1 else result
+        return bytes.fromhex(result)

```

Comment:
`self.to_bytes((self.bit_length() + 7) // 8, byteorder='big')`, or just returning `self` would be enough.
https://docs.python.org/3/library/stdtypes.html#int.to_bytes

Reply:
It seems we don't need this method.